x-pack/plugins/security_solution/public/common/components/event_details/overview/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { StatusPopoverButton } from '../overview/status_popover_button';
+import { SeverityBadge } from '../../../../../public/detections/components/rules/severity_badge';
+
+export const NotGrowingFlexGroup = euiStyled(EuiFlexGroup)`
+  flex-grow: 0;
+`;
+
+interface Props {
+  browserFields: BrowserFields;
+  contextId: string;
+  data: TimelineEventsDetailsItem[];
+  eventId: string;
+  handleOnEventClosed: () => void;
+  indexName: string;
+  timelineId: string;
+}
+
+export const Overview = React.memo<Props>(
+  ({ browserFields, contextId, data, eventId, handleOnEventClosed, indexName, timelineId }) => {
+    const statusData = useMemo(() => {
+      const item = find({ field: SIGNAL_STATUS_FIELD_NAME, category: 'kibana' }, data);
+      return (
+        item &&
+        getEnrichedFieldInfo({
+          eventId,
+          contextId,
+          timelineId,
+          browserFields,
+          item,
+        })
+      );
+    }, [browserFields, contextId, data, eventId, timelineId]);
+
+    const severityData = useMemo(() => {
+      const item = find({ field: 'kibana.alert.rule.severity', category: 'kibana' }, data);
+      return (
+        item &&
+        getEnrichedFieldInfo({
+          eventId,
+          contextId,
+          timelineId,
+          browserFields,
+          item,
+        })
+      );
+    }, [browserFields, contextId, data, eventId, timelineId]);
+
+    const riskScoreData = useMemo(() => {
+      const item = find({ field: 'kibana.alert.rule.risk_score', category: 'kibana' }, data);
+      return (
+        item &&
+        getEnrichedFieldInfo({
+          eventId,
+          contextId,
+          timelineId,
+          browserFields,
+          item,
+        })
+      );
+    }, [browserFields, contextId, data, eventId, timelineId]);
+
+    const ruleNameData = useMemo(() => {
+      const item = find({ field: SIGNAL_RULE_NAME_FIELD_NAME, category: 'kibana' }, data);
+      const linkValueField = find({ field: 'kibana.alert.rule.uuid', category: 'kibana' }, data);
+      return (
+        item &&
+        getEnrichedFieldInfo({
+          eventId,
+          contextId,
+          timelineId,
+          browserFields,
+          item,
+          linkValueField,
+        })
+      );
+    }, [browserFields, contextId, data, eventId, timelineId]);
+
+    return (
+      <NotGrowingFlexGroup gutterSize="s">
+        {hasData(statusData) && (
+          <EuiFlexItem>
+            <OverviewCard title={SIGNAL_STATUS}>
+              <StatusPopoverButton
+                eventId={eventId}
+                contextId={contextId}
+                enrichedFieldInfo={statusData}
+                indexName={indexName}
+                timelineId={timelineId}
+                handleOnEventClosed={handleOnEventClosed}
+              />
+            </OverviewCard>
+          </EuiFlexItem>
+        )}
+
+        {hasData(severityData) && (
+          <EuiFlexItem>
+            <OverviewCardWithActions
```

Comment:
When run locally, the actions were observed to sometimes cause `Severity` to be truncated, per the screenshot below:

![severity_truncation](https://user-images.githubusercontent.com/4459398/144663887-d0f64330-1f90-4acf-a2e9-6cc54c087eb4.png)

and sometimes only display a dot, per the screenshot below:

![severity_hidden](https://user-images.githubusercontent.com/4459398/144664077-5dac40a8-b17b-4236-b413-c0565e1be610.png)

When (for desk testing purposes) `OverviewCardWithActions` is temporarily changed to `OverviewCard`, the truncation / hidden behavior doesn't happen.

Reply:
consider desk-testing any layout-related changes in both the `Security > Alerts` page and separately in Timeline, because per the screenshot below, Timelines (non-resizable) details panel will grow and shrink as the browser is resized:

![alert_details_timeline](https://user-images.githubusercontent.com/4459398/144667586-58b2252f-85d1-4df1-81ea-dfe77ec26f2f.png)

Reply:
Great find, I adressed those issues in https://github.com/elastic/kibana/pull/120347/commits/d7da39d5bbb7c07b10ea30ce8235f2fd5ccab559 and https://github.com/elastic/kibana/pull/120347/commits/d33a0ab0e0177de8feb1aff9d18d0ef4c5c86d37. Also check out my comment explaining the new solution: https://github.com/elastic/kibana/pull/120347#issuecomment-987918713